pkg/compactor/block_upload.go

```diff
@@ -184,19 +195,25 @@ func (c *MultitenantCompactor) parseBlockUploadParameters(r *http.Request) (ulid
 	return blockID, tenantID, nil
 }
 
-func writeBlockUploadError(err error, op, extra string, logger log.Logger, w http.ResponseWriter) {
+func writeBlockUploadError(err error, msg string, logger log.Logger, w http.ResponseWriter) {
 	var httpErr httpError
 	if errors.As(err, &httpErr) {
-		level.Warn(logger).Log("msg", httpErr.message, "operation", op)
+		level.Warn(logger).Log("msg", msg, "response", httpErr.message, "status", httpErr.statusCode)
 		http.Error(w, httpErr.message, httpErr.statusCode)
 		return
 	}
 
-	if extra != "" {
-		extra = " " + extra
-	}
-	level.Error(logger).Log("msg", fmt.Sprintf("an unexpected error occurred%s", extra), "operation", op, "err", err)
-	http.Error(w, "internal server error", http.StatusInternalServerError)
+	// If customer complains with "I got an internal server error" give them something we could correlate to our logs.
+	id := hexTimeNowNano()
+	level.Error(logger).Log("msg", msg, "err", err, "error_id", id)
```

Comment:
Shall we add that to the logger (not as `error_id` but say `request_id`)? That way all log messages related to the same request will have same request ID and we can hopefully better troubleshoot what happened.

Reply:
Good idea, added request_id to all loggers in https://github.com/grafana/mimir/pull/5711/commits/07d219d16a2f149b80ab172e6cdb66af936e222f